### PR TITLE
WP-966 expose appid

### PR DESCRIPTION
**Pull request by ccameroni**
http://jira.webinos.org/browse/WP-966

Modified sessionWebApp to use installId when widget is running into the widget renderer.

### CODE REVIEW

File: lib/pzp_websocket.js [hunk 1]
```diff
@@ -460,9 +460,21 @@ function PzpWebSocketServer(){
...
-            var sessionWebApp = require("crypto").createHash("md5").update(PzpObject.getSessionId() + webAppName).digest("hex");
-              appId = PzpObject.getSessionId()  + "/"+ sessionWebApp +"_" + Math.round(Math.random()*100);
+            var sessionWebApp = null;
+            if (!webAppName) {
+                webAppName = require("crypto").randomBytes(3).toString("hex").toUpperCase();
+            } else if (webAppName.indexOf('widget') == 1) {
+                // widget's path syntax:
+                // /widget/<install_id>/<session_id>/file
+                var items = webAppName.split('/');
+                if (items.length == 5) {
```
Comment (TobyEalden) [inline lib/pzp_websocket.js:470]: I don't think you can hard-code checking for 5 components in the path - as widgets may contain content in sub-folders, e.g. /widget/<install_id>/<session_id>/folder1/folder2/index.html

It's probably sufficient that the path has /widget/<install_id>/<session_id> components.

**Comment by habibvirji** [comment]
Sorry cesare, I have just re-organised code. I could merge your changes but then my changes might affect your changes soon. 

If possible can you do your changes to my branch. I can include my changes along with your changes, please modify here or else if you could wait for a day or two I will merge my changes and then you can check if everything works for you then: https://github.com/habibvirji/webinos-pzp/blob/master/lib/pzp_id.js#L98

**Comment by ccameroni** [comment]
@TobyEalden 
OK, I will remove this check.

@habibvirji 
I can wait for the merge of your changes.

Thanks

**Comment by habibvirji** [comment]
@ccameroni Please resubmit this PR.. I am closing this PR for timebeing.

**Pull request closed without merging**